src/app/components/SearchModal.tsx

```
@@ -0,0 +1,301 @@
+
+    window.addEventListener('keydown', handleKeyDown);
+    return () => window.removeEventListener('keydown', handleKeyDown);
+  }, []);
+
+  const handleNavigate = useCallback((type: string, id: string) => {
+    switch (type) {
+      case 'book':
+        router.push(`/book/${id}`);
+        break;
+      case 'transaction':
+        const tx = results.transactions.find((t) => t.id === id);
```

Biome (2.4.9)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)